### DISCUSSION TITLE
Support Github's `noreply` email addresses

### DISCUSSION THREAD

**Discussion by amomchilov** [Automattic/gravatar#117] [Ideas]
E.g. `amomchilov@users.noreply.github.com` [docs](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-no-reply-email)

These e-mail addresses are impossible to verify to Gravatar (because GitHub doesn't forward the emails). As a result, commits made on GitHub's UI don't show a profile picture for any Git clients that use Gravatars for their profile pictures.

<img width="277" alt="image" src="https://github.com/user-attachments/assets/db4b812b-a988-4fe5-a23e-8e9a3113b96e">

**Comment by aaronfc** [comment]
Because of how Gravatar works this is not an easy task to do.
Ideally Github would make emails delivered to that address available to the user and Gravatar could work with normality.

Nevertheless, I have created an internal ticket to explore if we could use our existing Github account verification integration to do something around that. But truth be told this is a big effort and probably low impact in the service.

**Reply by opr** [reply]
The GitHub API emails endpoint thankfully includes the `@users.noreply.github.com` email, I was worried it would only include the top level "user" emails.

> I have created an internal ticket to explore if we could use our existing Github account verification integration to do something around that.

Awesome :) sounds like a solution that could work, though I concede I don't know the internals of how Gravatar works.

The process in my humble opinion is: when the account is linked (and possibly at set periods after it) the _verified_ emails could be added to the Gravatar account.

It may even be desirable to only add the `@users.noreply.github.com` emails to Gravatar and leave the other emails to be added through the normal process.

How the emails response looks:

```
{
	"email": "thomas.roberts@automattic.com",
	"primary": false,
	"verified": true,
	"visibility": null
},
{
	"email": "5656702+opr@users.noreply.github.com",
	"primary": false,
	"verified": true,
	"visibility": null
}
```

**Reply by seompark** [reply]
Any updates?

**Reply by Outlawdogg** [reply]
Hi my avatar I use is a pitbull dogg wearing a black leather trench , Black leather pants, Boots , a white T shirt says outlaw Dogg A skull belt buckle
Black cowboy hat with skull on it

**Reply by philostler** [reply]
Did this ever go anywhere?
